src/Utilities/Compiler/Extensions/IOperationExtensions.cs

```diff
@@ -512,6 +512,10 @@ public static bool IsWithinLambdaOrLocalFunction(this IOperation operation)
         {
             return pattern switch
             {
+#if CODEANALYSIS_V3_OR_BETTER
```

Comment:
Did you mean to target 2.9.x?

Reply:
Ah yes, thanks!

Reply:
Actually, no. These new IOperation APIs have been introduced in Microsoft.CodeAnalysis v3.3

Reply:
Maybe it should be named CODEANALYSIS_V3_3_OR_BETTER? Dunno where that's defined.

Reply:
Sure, I can clean that up in a separate PR once this goes in. That works?

Reply:
@dotpaul Any further feedback?